tests/components/automation/test_init.py

```diff
@@ -2771,6 +2771,7 @@ async def mock_stop_scripts_at_shutdown(*args):
                     ],
                     "action": [
                         {"service": "test.automation_started"},
+                        {"delay": 0.001},
```

Comment:
If there is no delay here, the infinite loop will never yield so it will never finish otherwise. This is apparently intended to be allowed as restart mode automations are allowed to restart themselves